File: Python/BEGINNER/1957 Converting to Hexadecimal.py
# SUBPROGRAM
def hex(n):
    array = list()
    while n >= 1:
        array += [int(n % 16)]
        n /= 16
    array = array[::-1]
    index = len(array) - 1
    for number in array:
        if number < 10:
            print(number, end='' if index != 0 else None)
        elif number == 10:
            print("A", end='' if index != 0 else None)
        elif number == 11:
            print("B", end='' if index != 0 else None)
        elif number == 12:
            print("C", end='' if index != 0 else None)
        elif number == 13:
            print("D", end='' if index != 0 else None)
        elif number == 14:
            print("E", end='' if index != 0 else None)
        else:
            print("F", end='' if index != 0 else None)
        index -= 1
    return None

# MAIN PROGRAM
num = int(input())
hex(num)

File: C/BEGINNER/1957 Converting to Hexadecimal.c
#include <stdio.h>

void hex(long int n);

int main()
{
    long int num;
    scanf("%ld", &num);
    hex(num);
    return 0;
}

void hex(long int n)
{
    int len = 0; 
    int array[1000];
    while (n != 0)
    {
        array[len] = (n % 16);
        len += 1;
        n /= 16;
    }
    
    for (int i = len - 1; i >= 0; i--)
    {
        if (array[i] < 10)
            printf("%d", array[i]);
        else if (array[i] == 10)
            printf("A");
        else if (array[i] == 11)
            printf("B");
        else if (array[i] == 12)
            printf("C");
        else if (array[i] == 13)
            printf("D");
        else if (array[i] == 14)
            printf("E");
        else
            printf("F");
    }
    printf("\n");
}